crates/common/src/local_db/fetch.rs

```diff
@@ -269,44 +289,28 @@ impl LocalDb {
     }
 }
 
-const RETRY_DELAY_MILLIS: u64 = 100;
-
 #[derive(Clone)]
 struct LogFetchJob {
     address: Address,
     range: BlockRange,
 }
 
-async fn retry_with_attempts<T, F, Fut>(
-    operation: F,
-    max_attempts: usize,
-) -> Result<T, LocalDbError>
-where
-    F: Fn() -> Fut,
-    Fut: std::future::Future<Output = Result<T, RpcClientError>>,
-{
-    if max_attempts == 0 {
-        return Err(LocalDbError::Config {
-            message: "max_attempts must be > 0".to_string(),
-        });
+fn map_retry_error(error: RetryError<LocalDbError>) -> LocalDbError {
```

Comment:
hm this also looks like a novel way to nest errors. we don't do this elsewhere in the codebase